scripts/method3.r

```diff
@@ -0,0 +1,154 @@
+## Method 3
+library(tweedie) 
+library(ggplot2)
+library(doParallel)
+library(foreach)
+
+simTweedieTest <-  
+  function(N){ 
+    t.test( 
+      rtweedie(N, mu=10000, phi=100, power=1.9), 
+      mu=10000 
+    )$p.value 
+  } 
+
+# Define MTweedieTests function with parallel execution
+MTweedieTests <- function(N, M, sig) {
+  results <- foreach(i = 1:M, .combine = '+') %do% {
+    simTweedieTest(N) < sig
+  }
+  
+  return(results / M)
+}
+
+# Set up parallel processing using doParallel
+cl <- makeCluster(detectCores())
+registerDoParallel(cl)
+
+df <- expand.grid(
+  N = c(10, 100, 1000, 5000, 10000),
+  M = 1000,
+  share_reject = NA
+)
+
+# Parallelize the computation using MTweedieTests function
+df$share_reject <- foreach(i = 1:nrow(df), .combine = 'c') %do% {
+  MTweedieTests(N = df$N[i], M = df$M[i], sig = 0.05)
+}
+
+# Stop the parallel workers
+stopCluster(cl)
+
+
+# Assignment 3:  
+df <-  
+  expand.grid( 
+    N = c(10,100,1000,5000, 10000), 
+    M = 1000, 
+    share_reject = NA) 
+
+
+for(i in 1:nrow(df)){ 
```

Comment:
Here you apply the function twice in the same script. Then it will overwrite what was there before.